ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,17 +10,17 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
-
     def initialize(source_or_path_or_io)
       @line_number = 0
-      if String === source_or_path_or_io
-        if File.file?(source_or_path_or_io)
-          @io = File.open(source_or_path_or_io, 'r:BOM|UTF-8')
-        else
-          @io = StringIO.new(source_or_path_or_io)
-        end
-      else
+
+      if File.file?(source_or_path_or_io.to_s)
```

Comment:
I've changed my mind about passing a path. I don't think we should allow that (none of the other implementations do). Let's remove it altogether! (and rename the variable to `source_or_io`